### PR TITLE
Add functions to cast tensor to different kinds

### PR DESCRIPTION
**Pull request by trfdeer**
### Checklist

- [x] Confirm that `run-checks.sh` has been executed.

### Related Issues/PRs

Related Issue: #487 

### Changes

1. Added `.int()` method to convert float tensors to integer kind.
2. Added `.float()` method to convert int tensors to float kind.

### Testing

Tested by manually creating and converting tensors.

### CODE REVIEW

**Comment by antimora** [comment]
Great. Thank you. We will need some unit tests because we can accept the PR. 

Please see this recent PR for how to add a test: https://github.com/burn-rs/burn/pull/486/files

There are 3 parts: 

1. add test under tests/ops. Make sure the file name is `cast` and the macro on the top file includes the name: like this `#[burn_tensor_testgen::testgen(cast)]`
2. add the auto generated test here: burn-tensor/src/tests/mod.rs
3. expose the module here: burn-tensor/src/tests/ops/mod.rs

To test, you'll need to cd to burn-ndarray and run `cargo test`. The test will use burn-ndarray backend. `run-checks.sh` will test on all backends.

File: burn-tensor/src/tensor/api/float.rs [hunk 1]
```diff
@@ -103,6 +104,24 @@ where
...
+    /// use burn_tensor::Tensor;
+    ///
+    /// fn example<B: Backend>() {
+    ///     let float_tensor = Tensor::<B, 1>::from_floats([1.0, 2.0]);
+    ///     let int_tensor = float_tensor.int();
+    /// }
+    /// ```
+    pub fn int(&self) -> Tensor<B, D, Int> {
+        Tensor::<B, D, Int>::from_data(self.to_data().convert())
+    }
```
Comment (nathanielsimard) [inline burn-tensor/src/tensor/api/float.rs:123]: I would take ownership here and call `into_data`. In a following PR we should call a backend function where each backend can decide how they do the conversion. For instance, in this current implementation, the wgpu backend would need to allocate at least 3 new buffers, where 0 or 1 could have been allocated with proper casting.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
A little change to the API before we can merge it. Thanks!

**Review by antimora — APPROVED** [review]
Looks good. Thank you for your contribution.

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**